### PR TITLE
Bhbc 2045 secure documents prototype

### PR DESCRIPTION
**Pull request by anissa-agahchen**
# Overview

## Links to Jira tickets

-WIP stable chunk of code that can be merged into the secure-documents-prototype

### CODE REVIEW

File: api/src/models/project-survey-attachments.ts [hunk 1]
```diff
@@ -67,15 +76,18 @@ export class GetReportAttachmentMetadata {
   description: string;
   year_published: number;
   revision_count: number;
+  security_review_timestamp: string;
```
Comment (curtisupshall) [inline api/src/models/project-survey-attachments.ts:79]: Let's make a `security` property which stores `security_review_timestamp` and `security_reasons`.

File: app/src/components/dialog/ViewFileWithDetailsDialog.tsx [hunk 2]
```diff
@@ -48,27 +52,28 @@ const useStyles = makeStyles((theme: Theme) => ({
...
 
 /**
  * General information content for a project.
  *
  * @return {*}
  */
-const ViewFileWithMetaDialog: React.FC<IViewFileWithMetaDialogProps> = (props) => {
+const ViewFileWithDetailsDialog: React.FC<IViewFileWithDetailsDialogProps> = (props) => {
   const classes = useStyles();
-  const { reportMetaData } = props;
```
Comment (curtisupshall) [inline app/src/components/dialog/ViewFileWithDetailsDialog.tsx:68]: Add this back in?

File: app/src/components/dialog/ViewFileWithDetailsDialog.tsx [hunk 3]
```diff
@@ -196,28 +169,36 @@ const ViewFileWithMetaDialog: React.FC<IViewFileWithMetaDialogProps> = (props) =
...
-                    <TableCell></TableCell>
-                  </TableRow>
+                  {props.reportMetaData?.security_reasons &&
+                    props.reportMetaData?.security_reasons?.length > 0 &&
+                    props.reportMetaData?.security_reasons?.map((row, index) => {
+                      return (
+                        <TableRow key={`${row.category}-${index}`}>
+                          <TableCell>{row.category}</TableCell>
+                          <TableCell>
+                            <Typography style={{ fontWeight: 700 }}>{row.reason}</Typography>
```
Comment (curtisupshall) [inline app/src/components/dialog/ViewFileWithDetailsDialog.tsx:182]: Make this a class?

File: app/src/interfaces/useProjectApi.interface.ts [hunk 4]
```diff
@@ -295,13 +296,22 @@ export interface IGetReportMetaData {
   last_modified: string;
   revision_count: number;
   authors: IGetReportAuthors[];
+  security_reasons: IGetSecurityReasons[];
```
Comment (curtisupshall) [inline app/src/interfaces/useProjectApi.interface.ts:299]: Make this a `security` property

File: app/src/interfaces/useSurveyApi.interface.ts [hunk 5]
```diff
@@ -170,6 +170,7 @@ export interface IGetSurveyAttachment {
   lastModified: string;
   size: number;
   securityToken: any;
+  securityReviewTimestamp: string;
```
Comment (curtisupshall) [inline app/src/interfaces/useSurveyApi.interface.ts:173]: make security parameter

File: app/src/interfaces/useProjectApi.interface.ts [hunk 6]
```diff
@@ -14,6 +14,7 @@ export interface IGetProjectAttachment {
   lastModified: string;
   size: number;
   securityToken: string;
+  securityReviewTimestamp: string;
```
Comment (curtisupshall) [inline app/src/interfaces/useProjectApi.interface.ts:17]: security parameter

File: app/src/interfaces/useSurveyApi.interface.ts (hunk 5 above)
Comment (NickPhura) [reply app/src/interfaces/useSurveyApi.interface.ts:173]: Might need to elaborate on what you mean.

I think this looks correct, unless I'm forgetting something. The database column this is mapping to is `security_review_timestamp`

Comment (NickPhura) [reply app/src/interfaces/useSurveyApi.interface.ts:173]: Oh, unless you mean putting the security stuff in its own object like we were talking about? That definitely makes sense when it comes to the list of all security rules applied to an attachment, but this particular column is part of the attachment table, so it makes sense to live in this object.

I think ultimately, we still want to maintain a separate object that represents a single attachment record, and a separate object that represents the rules applied to a single attachment. That way we can fetch them separately via services, etc. And then our attachment related endpoints can mix and match the service calls to return attachment(s) with or without security rules alongside them.

**Pull request merged**